src/convertToObject.js

```diff
@@ -6,7 +6,21 @@
  * @return {object}
  */
 function convertToObject(sourceString) {
-  // write your code here
+  const myArray = sourceString.split(';');
+
+  const tempArray = myArray.map((element, index, array) => {
+    const trimmedElement = element.trim();
+    const [key, value] = trimmedElement.split(':').map((item) => item.trim());
```

Comment:
The code assumes that every element in the `sourceString` is formatted as 'key:value'. If an element does not contain a colon, `split(':')` will return an array with a single element, causing `key` to be assigned a value and `value` to be `undefined`. Consider adding validation to handle cases where the format is incorrect.